### PR TITLE
add Legendre transform and suspension for lazy symmetric functions

### PR DESCRIPTION
**Pull request by fchapoton**
This adds two useful functions, particularly important when working with operads and Koszul duality.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/rings/lazy_series.py [hunk 1]
```diff
@@ -6673,6 +6675,60 @@ def coefficient(n):
...
+
+        Given a symmetric function `f` of valuation 2, the Legendre
+        transform of `f` is the unique symmetric function `g` of
+        valuation 2 over the same base ring, such that
+
+        .. MATH::
+
+            g \circ \partial_{p_1} f + f = p_1 \partial_{p_1} f.
+
+        This implies that the derivatives of `f` and `g` w.r.t. `p_1`
```
Comment (tscrim) [inline src/sage/rings/lazy_series.py]:
```suggestion
        This implies that the derivatives of `f` and `g` with respect to `p_1`
```
Good to spell it out IMO.

**Review by tscrim — APPROVED** [review]
LGTM modulo one trivial doc thing. You can set a positive review once done.

@mantepse cc-ing you in case you have any comments.

**Comment by mantepse** [comment]
Fine with me!

**Review by mantepse — APPROVED** [review]

**Comment by fchapoton** [comment]
Thanks a lot, Martin and Travis.

**Pull request merged**